index.js

```diff
@@ -356,3 +356,27 @@ for (let i = 1; i < 2; i += 1) {
 
   popupBody.appendChild(popupListDesktop);
 }
+
+const form = document.querySelector('form');
+const email = document.getElementById('email');
+const small = document.querySelector('small');
+
+form.addEventListener('submit', (event) => {
+  if (email.validity.valueMissing) {
+    small.innerHTML = 'Please enter an e-mail address';
+    event.preventDefault();
+    small.classList.remove('collapse');
+  } else if (email.validity.typeMismatch) {
+    small.innerHTML = 'E-mail address needs to be valid (example@org.com)';
+    event.preventDefault();
+    small.classList.remove('collapse');
+  } else if (email.validity.patternMismatch) {
+    small.innerHTML = 'E-mail address needs to be in lower case';
+    event.preventDefault();
```

Comment:
- [ ] Great job implementing the email validation logic. However, in other to maintain the DRY principle for javascript best practice, I will suggest you refactor the highlighted code to remove the `event.preventDefault()` from the if statement and place it at the top so we can use it once. Or you can achieve it whichever way. So long as we are not repeating it inside the function.